sycl/include/CL/sycl/detail/buffer_impl.hpp

```diff
@@ -35,6 +35,7 @@ class buffer;
 template <typename DataT, int Dimensions, access::mode AccessMode>
 class host_accessor;
 
+template <typename T = char>
```

Comment:
After looking at the StackOverflow link this should globally work according to thelink provided towards the end giving a code like this I have extended a little bit:
```c++
#include <iostream>

template <typename T=int>
class tt {
public:
  T get() { return 5; }
};

int main()
{
  tt t;  // avoid <>
  std::cout << t.get() << std::endl;
  tt<> ttt;
  std::cout << ttt.get() << std::endl;
}
```
But I think the problem here is that you are using this type alias here which does not work with CTAD before C++20.
While I would love to just switch to C++20 :-), is there a way to short-cut this type alias?

Reply:
@keryell, it looks like the code works fine for C++17 but has the issue for C++14 mode.
@nawrinsu could you please provide an implementation that works both for C++17 and C++14. Default mode is C++17 but we still support C++14 when possible).

Reply:
C++14 is le passé : 7 year old... SYCL is the future! :-)

Reply:
@vladimirlaz I'll update the PR soon.